mbuild/formats/lammpsdata.py

```diff
@@ -856,29 +550,85 @@ def write_lammpsdata(
                         )
                     )
 
+    return
+
+
+def _evaluate_lj_conversion_factors(
+    structure, conversion_name, conversion_factor
+):
+    """Get Lennard Jones style conversion factors. `conversion_name`` can be sigma, epsilon, or mass."""
+    if conversion_factor is None:
+        # Check if structure is parametrized
+        if any([atom.sigma for atom in structure.atoms]) is None:
+            raise ValueError(
+                "LJ units specified but one or more atoms has undefined LJ "
+                "parameters."
+            )
+        else:
+            conversion_factor = np.max(
+                [getattr(atom, conversion_name) for atom in structure.atoms]
+            )
+            warn(
+                f"Assuming {conversion_name} conversion factor of "
+                + str(conversion_factor)
+            )
+        if conversion_factor == 0:
+            conversion_factor = 1
+            warn(
+                f"{conversion_name} conversion factor cannot be inferred from the maximum {conversion_name} value in the ParmEd Structure. "
+                "Setting the {conversion_name} conversion factor to 1"
+            )
+    elif conversion_factor <= 0:
+        raise ValueError(
+            f"The {conversion_name} conversion factor to convert to LJ units should be greater than 0."
+        )
+    else:
+        # assume conversion factor passed in mbuild units, convert to lammps real units
+        if conversion_name == "sigma":
+            conversion_factor *= NM_TO_ANG
```

Comment:
so if I understand correctly, if a user sees a warning above saying the sigma conversion factor is set to 1, the actual mathematically value of the sigma conversion factor will be 10, correct? seems a bit confusing. Maybe better to just move the warnings below this, after all the conversions have been settled.

Reply:
The mathematical sigma conversion factor is 1.0 nm. Which is what is passed in in mBuild units. I just convert everything to lammps units as soon as they're read in, so non-dimensionalizing everything later on is straightforward. So mathematically it is 1.0 nm or 10.0 angstrom, but they're the same number.